scripts/service-accounts/generateServiceAccounts.go

```diff
@@ -38,42 +45,130 @@ var (
 		"dp-api-router",
 		"dp-download-service",
 		"dp-frontend-geography-controller",
-		"dp-identity-api"}
+		"dp-identity-api",
+	}
 )
 
+type ServiceAccount struct {
+	ID   string `json:"-"`
+	Name string `json:"id,omitempty"`
+}
+
+func init() {
+	s1 := rand.NewSource(time.Now().UnixNano())
+	r = rand.New(s1)
+}
+
 func main() {
+	serviceDir := flag.String("dir", "service", "the service account directory path")
+	flag.Parse()
+
+	if err := createServiceDir(*serviceDir); err != nil {
+		log.Event(nil, "error checking service dir", log.Error(err))
+		os.Exit(1)
+	}
+
+	existingAccounts, err := loadExisting(*serviceDir)
+	if err != nil {
+		log.Event(nil, "error loading existing service accounts", log.Error(err))
+		os.Exit(1)
+	}
 
 	for _, service := range services {
+		logD := log.Data{"service": service}
 
-		account := map[string]interface{}{
-			"id": service,
+		if _, ok := existingAccounts[service]; ok {
+			log.Event(nil, "a service account already exists for service skipping ", logD)
+			continue
 		}
 
-		id := newRandomID(64)
-
-		jsonData, err := json.Marshal(account)
+		acc, err := createServiceAccountJSON(*serviceDir, service)
 		if err != nil {
-			fmt.Sprintf("Failed when attempting to marshal json for service: ", service)
-			panic(err)
+			log.Event(nil, "error creating service account", log.Error(err), logD)
+			os.Exit(1)
 		}
+		fmt.Printf("%s\t %s\n", acc.ID, acc.Name)
+	}
+
+	log.Event(nil, "successfully created service accounts")
+}
+
+func createServiceAccountJSON(path, service string) (*ServiceAccount, error) {
+	serviceAcc := &ServiceAccount{
+		ID:   newRandomID(64),
+		Name: service,
+	}
 
-		jsonFile, err := os.Create("./" + id + ".json")
+	logD := log.Data{
+		"service": service,
+	}
+
+	jsonData, err := json.Marshal(serviceAcc)
+	if err != nil {
+		log.Event(nil, "error attempting to marshal service account to json", log.Error(err), logD)
+		return nil, err
+	}
+
+	p := filepath.Join(path, serviceAcc.ID+".json")
+	jsonFile, err := os.Create(p)
+	if err != nil {
+		log.Event(nil, "error creating service account json", log.Error(err), logD)
```

Comment:
Consider including the word 'file' in this log message to remove ambiguity with marshalling (if you didn't know what the contents of the marshal error looked like, you might read this and not know which of the two it was)